### ISSUE TITLE
Request templates

### ISSUE THREAD

**Issue by keithbrink** [kviklet/kviklet#163]
It would be really nice if we could have parametrized templates so that our CS team can pick the correct template, fill in the variables, and the full query would be added as a request.

Imagine something like this:

`UPDATE users SET name = $NAME WHERE id = $ID`

**Comment by Askir** [comment]
This goes into a direction of internal tooling builder which is already quite the saturated field.
There is [retool](https://retool.com/), and [windmill ](https://github.com/windmill-labs/windmill)(open-source) for this purpose. There of course also used to be airplane.dev but they shut down earlier this year.
I highly recommend to go with one of these tools if your use-case is building internal tooling for known issues.

I intended Kviklet to be for the exceptional database access use-case , which you run into every now and then when something is broken and  you don't have a process yet to fix it. Usually this is done by an engineer that knows SQL.

However I got the request of configurable templates multiple times now and there seems to be a bit of a gap between "I want some database queries to be run" and "I want to build internal tooling for my CS team". So I think i'll reconsider this and see if I can fill that gap.

Btw, you can already copy existing requests, so if you're comfortable with it a decent workflow would also be:
1. Link an existing request with `UPDATE users SET name = $NAME WHERE id = $ID`
2. Copy from this request
3. Edit the query and change the ID only (Your non technical users would have to be comfortable with this instead of a text field)
4. Create the new Request

With more complex queries this is of course not ideal but you could probably use your database engines variables to make it easier by moving all the "templating-options" to the top of the statement.

___

Just some ideas on how to do this for now. I will look into how to do templating though! I think it fits the tool quite well.

**Comment by keithbrink** [comment]
Makes sense, and fully support keeping Kviklet focused on the exceptional access use case, as that's our primary reason for using it as well. Once I start using a tool though, I don't like adding another one for a related but slightly different use case, but totally understand if you feel that is scope creep.

**Comment by Askir** [comment]
No this is totally fair.
I just started with the need for something that isn't a fully fleshed internal tooling builder and also not an analytics tool like metabase and I am used to people confusing it for those, so I am a bit hesitant with expanding in those directions.
But you're right that some adjacent use cases make perfect sense to cover.

**Comment by othmane-kinane-nw** [comment]
this is a use case I'd love to see too
